core/tee/tee_pobj.c

```diff
@@ -37,20 +37,21 @@ TAILQ_HEAD_INITIALIZER(tee_pobjs);
 static TEE_Result tee_pobj_check_access(uint32_t oflags, uint32_t nflags)
 {
 	/* meta is exclusive */
-	if (oflags | TEE_DATA_FLAG_ACCESS_WRITE_META ||
-	    nflags | TEE_DATA_FLAG_ACCESS_WRITE_META)
-		return TEE_ERROR_ACCESS_CONFLICT;
-
-	if (oflags | TEE_DATA_FLAG_ACCESS_READ &&
-	    !((nflags | TEE_DATA_FLAG_SHARE_READ) &&
-	      oflags | TEE_DATA_FLAG_SHARE_READ))
-		return TEE_ERROR_ACCESS_CONFLICT;
-
-	if (oflags | TEE_DATA_FLAG_ACCESS_WRITE &&
-	    !((nflags | TEE_DATA_FLAG_SHARE_WRITE) &&
-	      oflags | TEE_DATA_FLAG_SHARE_WRITE))
+	if (oflags & TEE_DATA_FLAG_ACCESS_WRITE_META)
 		return TEE_ERROR_ACCESS_CONFLICT;
 
+	/*second open/create need set flags to TEE_DATA_FLAG_SHARE_READ */
+	if (((oflags & TEE_DATA_FLAG_ACCESS_READ) ||
```

Comment:
And I think table 5-14 is just examples, line 7,said "An object can be opened with only share flags, which locks the access to an object against a given mode." is useless,redundancy  because  there is only two states(TEE_DATA_FLAG_SHARE_READ and TEE_DATA_FLAG_ACCESS_READ) for read,and TEE_DATA_FLAG_SHARE_READ to control  SHARE to use.